cores/esp8266/StreamDev.h

```diff
@@ -0,0 +1,255 @@
+    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+    parsing functions based on TextFinder library by Michael Margolis
+*/
+
+#ifndef __STREAMDEV_H
+#define __STREAMDEV_H
+
+#include <StreamString.h>
+
+///////////////////////////////////////////////
+// /dev/null
+// - black hole as output, swallow everything, availableForWrite = infinite
+// - black hole as input, nothing to read, available = 0
+
+class StreamNull: public Stream
+{
+public:
+
+    // Print
+    virtual size_t write(uint8_t) override
+    {
+        return 1;
+    }
+
+    virtual size_t write(const uint8_t* buffer, size_t size) override
+    {
+        (void)buffer;
+        (void)size;
+        return size;
+    }
+
+    virtual int availableForWrite() override
+    {
+        return 32767;
+    }
+
+    // Stream
+    virtual int available() override
+    {
+        return 0;
+    }
+
+    virtual int read() override
+    {
+        return -1;
+    }
+
+    virtual int peek() override
+    {
+        return -1;
+    }
+
+    virtual size_t readBytes(char* buffer, size_t len) override
+    {
+        (void)buffer;
+        (void)len;
+        return 0;
+    }
+
+    virtual int read(uint8_t* buffer, size_t len) override
+    {
+        (void)buffer;
+        (void)len;
+        return 0;
+    }
+
+    virtual bool outputTimeoutPossible() override
+    {
+        return false;
+    }
+
+    virtual bool inputTimeoutPossible() override
+    {
+        return false;
+    }
+
+    virtual ssize_t streamSize() override
+    {
+        return 0;
+    }
+};
+
+///////////////////////////////////////////////
+// /dev/zero
+// - black hole as output, swallow everything, availableForWrite = infinite
+// - big bang as input, gives infinity to read, available = infinite
+
+class StreamZero: public StreamNull
+{
+protected:
+
+    char _x;
+
+public:
+
+    StreamZero(char x = 0): _x(x) { }
+
+    // Stream
+    virtual int available() override
+    {
+        return 32767;
+    }
+
+    virtual int read() override
+    {
+        return _x;
+    }
+
+    virtual int peek() override
+    {
+        return _x;
+    }
+
+    virtual size_t readBytes(char* buffer, size_t len) override
+    {
+        memset(buffer, _x, len);
+        return len;
+    }
+
+    virtual int read(uint8_t* buffer, size_t len) override
+    {
+        memset((char*)buffer, _x, len);
+        return len;
+    }
+
+    virtual ssize_t streamSize() override
+    {
+        return 32767;
+    }
+};
+
+///////////////////////////////////////////////
+// static buffer (in flash or ram)
+// - black hole as output, swallow everything, availableForWrite = infinite
+// - Stream buffer out as input, resettable
+
+class StreamPtr: public StreamNull
+{
+protected:
+    const char* _buffer;
+    size_t _size;
+    bool _in_flash;
+    size_t _peekPointer = 0;
+
+public:
+    StreamPtr(const String& string): _buffer(string.c_str()), _size(string.length()), _in_flash(false) { }
```

Comment:
Can you remove the `in_flash` param completely and just check the actual address is <0x4000_0000?  One less thing for users to get wrong.

Reply:
done